test/parser.coffee

```diff
@@ -37,228 +37,253 @@ describe 'VASTParser', ->
+
+                it 'should have 1 URL for customclick', =>
+                    linear.videoCustomClickURLTemplates.should.eql ['http://example.com/linear-customclick']
+
+                it 'should have 8 tracking events', =>
+                    linear.trackingEvents.should.have.keys 'start', 'close', 'midpoint', 'complete', 'firstQuartile', 'thirdQuartile', 'progress-30', 'progress-60%'
+
+                it 'should have 3 URLs for start event', =>
+                    linear.trackingEvents['start'].should.eql ['http://example.com/linear-start', 'http://example.com/wrapperB-linear-start', 'http://example.com/wrapperA-linear-start']
+
+                it 'should have 3 URLs for complete event', =>
+                    linear.trackingEvents['complete'].should.eql ['http://example.com/linear-complete', 'http://example.com/wrapperB-linear-complete', 'http://example.com/wrapperA-linear-complete']
+
+                it 'should have 3 URLs for progress-30 event VAST 3.0', =>
+                    linear.trackingEvents['progress-30'].should.eql ['http://example.com/linear-progress-30sec', 'http://example.com/wrapperB-linear-progress-30sec', 'http://example.com/wrapperA-linear-progress-30sec']
+
+                it 'should have 3 URLs for progress-60% event VAST 3.0', =>
+                    linear.trackingEvents['progress-60%'].should.eql ['http://example.com/linear-progress-60%', 'http://example.com/wrapperB-linear-progress-60%', 'http://example.com/wrapperA-linear-progress-60%']
+
+                it 'should have parsed icons element', =>
+                    icon = linear.icons[0]
+                    icon.program.should.equal "ad1"
+                    icon.height.should.equal 20
+                    icon.width.should.equal 60
+                    icon.xPosition.should.equal "left"
+                    icon.yPosition.should.equal "bottom"
+                    icon.apiFramework.should.equal "VPAID"
+                    icon.offset.should.equal 15
+                    icon.duration.should.equal 90
+                    icon.type.should.equal "image/gif"
+                    icon.staticResource.should.equal "http://example.com/linear-icon.gif"
+                    icon.iconClickThroughURLTemplate.should.equal "http://example.com/linear-clickthrough"
+                    icon.iconClickTrackingURLTemplates.should.eql ["http://example.com/linear-clicktracking1", "http://example.com/linear-clicktracking2"]
+                    icon.iconViewTrackingURLTemplate.should.equal "http://example.com/linear-viewtracking"
+
+            #Companions
+            describe '2nd creative (Companions)', ->
+                companions = null
+
+                before (done) =>
+                    companions = _response.ads[0].creatives[1]
```

Comment:
Again ^^. `after()` hook. 😉 
The same pattern is used several times bellow. Please make sure to fix it everywhere. 
Thx.